corehq/apps/hqwebapp/management/commands/build_requirejs.py

```diff
@@ -70,7 +70,18 @@ def handle(self, **options):
         # Overwrite each bundle in resource_versions with the sha from the optimized version in staticfiles
         for module in config['modules']:
             filename = os.path.join(self.root_dir, 'staticfiles', module['name'] + ".js")
-            resource_versions[module['name'] + ".js"] = self.get_hash(filename)
+            file_hash = self.get_hash(filename)
+
+            # Overwrite source map reference. Source maps are accessed on the CDN,
+            # so they need to have the version hash appended.
+            with open(filename, 'r') as fin:
+                lines = fin.readlines()
+            with open(filename, 'w') as fout:
+                for line in lines:
+                    if re.search(r'sourceMappingURL=bundle.js.map', line):
+                        line = re.sub(r'bundle.js.map', 'bundle.js.map?version=' + file_hash, line)
+                    fout.write(line)
+            resource_versions[module['name'] + ".js"] = file_hash
```

Comment:
Should `get_hash(filename)` be called after the file is re-written (just above this line) since writing new content into the file will change the hash? I forget how these hashes are used, so not sure which is correct.

Also, could calculate the hash from the lines while reading it above, which may make the build a bit faster (eliminate a read). Not sure if this is worth it though.

Reply:
It's true that the hash used as the version isn't the true hash of the file because the content changes after `get_hash` is called. But there isn't a way around that because the file itself contains the hash: updating the file means a different hash, then updating to add the new hash means yet another hash, and so on.

But it's okay that the hash isn't "correct", all that matters is that the hash changes whenever the code changes. The fact that it changed at all is what forces the CDN to grab the newest code instead of a cached version.

Eliminating a read is a fair point, but I'm going to be lazy for now because testing this code is a pain, can revisit the next time I'm in this file.